holoviews/streams.py

```diff
@@ -149,7 +150,9 @@ def source(self):
     @source.setter
     def source(self, source):
         if self._source:
-            raise Exception('source has already been defined on stream.')
+            source_list = self.registry[id(self._source)]
+            if self in source_list:
+                source_list.remove(self)
```

Comment:
Overriding an existing ``source`` raised an Exception before, but I think it's fine as it's not something you'd do without knowing what you're doing anyway.